avocado/utils/software_manager.py

```diff
@@ -495,6 +499,39 @@ def provides(self, name):
             return None
 
 
+class DnfBackend(YumBackend):
+
+    """
+    Implements the dnf backend for software manager.
+
+    DNF is the successor to yum in recent Fedora.
+    """
+    def __init__(self):
+        """
+        Initializes the base command and the yum package repository.
+        """
+        super(DnfBackend, self).__init__()
+        executable = utils_path.find_command('dnf')
+        base_arguments = '-y'
+        self.base_command = executable + ' ' + base_arguments
+
+    def install(self, name):
+        """
+        Installs package [name]. Handles local installs.
+        """
+        command = 'install'
+        if os.path.isfile(name):
+            name = os.path.abspath(name)
```

Comment:
A question: have you had trouble with `dnf` using non absolute paths? I'm asking this because I'm looking at `YumBackend.install`, and it could really use this change:

```
diff --git a/avocado/utils/software_manager.py b/avocado/utils/software_manager.py
index 91647cf..0a3ef9d 100644
--- a/avocado/utils/software_manager.py
+++ b/avocado/utils/software_manager.py
@@ -387,12 +387,7 @@ class YumBackend(RpmBackend):
         """
         Installs package [name]. Handles local installs.
         """
-        if os.path.isfile(name):
-            name = os.path.abspath(name)
-            command = 'localinstall'
-        else:
-            command = 'install'
-
+        command = 'install'
         i_cmd = self.base_command + ' ' + command + ' ' + name
 
         try:
```

The reason being that `localinstall` has been deprecated for a while and `install` works just as well with local packages.

But, even if `dnf` need the absolute path, that can be added to `YumBackend` and then `DnfBackend` would be even a little slimmer (would loose `install`).

Makes sense?